internal/command/test.go

```diff
@@ -235,14 +211,37 @@ func (c *TestCommand) Run(rawArgs []string) int {
 			TestingDirectory:    args.TestDirectory,
 			Opts:                opts,
 			View:                view,
-			JUnit:               junitFile,
 			Stopped:             false,
 			Cancelled:           false,
 			StoppedCtx:          stopCtx,
 			CancelledCtx:        cancelCtx,
 			Filter:              args.Filter,
 			Verbose:             args.Verbose,
 		}
+
+		if args.JUnitXMLFile != "" {
+			// JUnit XML output is currently experimental, so that we can gather
+			// feedback on exactly how we should map the test results to this
+			// JUnit-oriented format before anyone starts depending on it for real.
+			if !c.AllowExperimentalFeatures {
+				diags = diags.Append(tfdiags.Sourceless(
+					tfdiags.Error,
+					"JUnit XML output is not available",
+					"The -junit-xml option is currently experimental and therefore available only in alpha releases of Terraform CLI.",
+				))
+				view.Diagnostics(nil, nil, diags)
+				return 1
```

Comment:
This early return is breaking the cancel and done contexts. This check might have to go back up - or you could merge the other PR removing this first and come back to this?

Reply:
Oops - I've moved the flag checks back before the cancellable contexts get created

> or you could merge the other PR removing this first and come back to this?

I can do that - _this_ PR can be merged before or after the feature leaves experimental status. The only time pressure is for the other PR